kmymoney: 5.1.1 → 5.1.3

das-g opened the pull request:
fixes #182499

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
- bump version to 5.1.3 ([announcement of 5.1.2](https://kmymoney.org/2021/06/23/kmymoney-5-1-2-released.html), [announcement of 5.1.3](https://kmymoney.org/2022/07/30/kmymoney-5-1-3-released.html))
  - this includes [a fix](https://invent.kde.org/office/kmymoney/-/commit/265d4dc3d1dc8592ec1122aa04bfb7d431a43d5d) for #182499 (upstream [Bug 443208](https://bugs.kde.org/show_bug.cgi?id=443208))
- replace `weboob` with `woob` everywhere in `pkgs/applications/office/kmymoney/default.nix`
- patch the WOOB Python script already in the source tree before building (in `postPatch`) rather than after building (`in postInstall`) in the build result `$out`, as it's now embedded into a binary rather than visible in the build result as a separate file.
- remove Python 2 from dependencies, as it isn't needed anymore
- make `libpython` available to `.kmymoney-wrapped`, so that the Python interpreter now embedded in that binary can work

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
  - [x] `meta.description` according to [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md)
  - [x] `meta.license` fits upstream license (confirmed by upstream [in their Telegram group](https://t.me/KMyMoney/10763))
  - [x] `meta.maintainers` set

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

das-g committed:
kmymoney: 5.1.1 → 5.1.3

aidalgol commented:
I found an [answer on StackOverflow](https://stackoverflow.com/questions/67891197/ctypes-cpython-39-x86-64-linux-gnu-so-undefined-symbol-pyfloat-type-in-embedd) that sounds relevant to the `ImportError` from the woob plugin.  I am still trying to fully understand the details myself, though, so it may be a similar symptom but a different root cause.

aidalgol commented:
I think we can safely drop `python2` from `nativeBuildInputs` now.

aidalgol committed:
kmymoney: Fix loading of woob plugin

Since kmymoney now embeds the python interpreter, we need to use patchelf to
make it able to find libpython at runtime.

aidalgol commented:
@das-g I have a fix for this PR, which I have submitted as a PR to the branch on your `nixpkgs` fork.

aidalgol committed:
kmymoney: Remove obsolete python2 build input

das-g commented:
Thanks, @aidalgol, your changes seem to have solved the remaining issues:
```
Loading "/nix/store/xgfc5pdnjpm18xi3f0pcc0nbnlm101qp-kmymoney-5.1.3/lib/qt-5.15.5/plugins/kmymoney/woob.so"
Python interpreter found: 3.10.5 (main, Jun  6 2022, 12:05:50) [GCC 11.3.0]
Saved a copy of the embedded :/plugins/woob/kmymoneywoob.py script as "/run/user/1000/kmymoneyImGAhg.py"
Attempt to load the "kmymoneyImGAhg" Python module from "/run/user/1000"
The dependency 'sys' Python module loaded successfully
The 'path' Python variable loaded successfully
"kmymoneyImGAhg" Python module loaded successfully
Plugins: woob loaded
```

This package currently has no maintainers listed. Would you be willing to maintain it together with me, @aidalgol?

SuperSandro2000 commented on pkgs/applications/office/kmymoney/default.nix:
```diff
@@ -62,6 +63,13 @@ stdenv.mkDerivation rec {
+  # libpython is required by the python interpreter embedded in kmymoney, so we
+  # need to explicitly tell autoPatchelf about it.
+  postFixup = ''
+    patchelf --debug --add-needed libpython3.10.so \
```
```suggestion
    patchelf --debug --add-needed libpython${python3.pythonVersion}.so \
```

das-g commented on pkgs/applications/office/kmymoney/default.nix:
```diff
@@ -38,20 +39,20 @@ stdenv.mkDerivation rec {
   ];
 
-  weboobPythonPath = [ python3Packages.weboob ];
+  woobPythonPath = [ python3Packages.woob ];
 
```
Will `buildPythonPath "$woobPythonPath"` [in `postPatch`](https://github.com/NixOS/nixpkgs/pull/183973/files#diff-4004fbc3fdb7ac27f994a39aaf8785f18889895120945e21fb984e7e338afc59R48) keep working without that?

SuperSandro2000 replied:
we are inling that in there

das-g replied:
applied with b371f5b5416643a42bdcd7fe8cf97dd07c191206

das-g commented on pkgs/applications/office/kmymoney/default.nix:
```diff
@@ -10,24 +11,24 @@
 # Needed for running tests:
 , qtbase, xvfb-run
 
```
```suggestion

, python3
```
(needed for https://github.com/NixOS/nixpkgs/pull/183973/files#r935577346)

das-g committed:
kmymoney: avoid hardcoding libpython version

(Apply suggestions from code review)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

SuperSandro2000 commented on pkgs/applications/office/kmymoney/default.nix:
```diff
@@ -10,24 +11,25 @@
 
-, python2, python3Packages
+, python3
+, python3Packages
```
```suggestion
```

das-g replied:
applied with 1234550f536a6f72069004b1cceb80fde3d8ffdc

SuperSandro2000 commented on pkgs/applications/office/kmymoney/default.nix:
```diff
@@ -38,20 +40,20 @@ stdenv.mkDerivation rec {
     # Put it into buildInputs so that CMake can find it, even though we patch
     # it into the interface later.
-    python3Packages.weboob
+    python3Packages.woob
```
```suggestion
    python3.pkgs.woob
```

das-g replied:
applied with 1234550f536a6f72069004b1cceb80fde3d8ffdc

SuperSandro2000 commented on pkgs/applications/office/kmymoney/default.nix:
```diff
@@ -38,20 +40,20 @@ stdenv.mkDerivation rec {
-    buildPythonPath "$weboobPythonPath"
-    patchPythonScript "$out/share/kmymoney/weboob/kmymoneyweboob.py"
+  postPatch = ''
+    buildPythonPath "$woobPythonPath"
```
```suggestion
    buildPythonPath "${python3.pkgs.woob}"
```

das-g replied:
applied with b371f5b5416643a42bdcd7fe8cf97dd07c191206 & 1234550f536a6f72069004b1cceb80fde3d8ffdc

aidalgol commented:
> This package currently has no maintainers listed. Would you be willing to maintain it together with me, @aidalgol?

Certainly.

das-g committed:
kmymoney: access python3Packages through python3.pkgs

and eliminate it from the function arguments

(Apply suggestions from code review)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

das-g committed:
kmymoney: inline woobPythonPath

(Apply suggestions from code review)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

das-g committed:
kmymoney: add maintainers aidalgol and das-g

aidalgol agreed here: https://github.com/NixOS/nixpkgs/pull/183973#issuecomment-1203225836

das-g commented:
Result of `nixpkgs-review pr 183973` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kmymoney</li>
  </ul>
</details>
